ofrak_components/ofrak_components/flash.py

```diff
@@ -0,0 +1,777 @@
+            tag = FlashBlock
+            if c == flash_attr.header_block_format:
+                tag = FlashHeaderBlock
+            elif c == flash_attr.tail_block_format:
+                tag = FlashTailBlock
+            await ecc_resource.create_child(
+                tags=(tag,),
+                data_range=block_range,
+            )
+
+            block_ecc_range = flash_attr.get_field_range_in_block(c, FlashFieldType.ECC)
+            if block_ecc_range is not None:
+                cur_block_ecc = block_data[block_ecc_range.start : block_ecc_range.end]
+                only_ecc += cur_block_ecc
+                # Add hash of everything up to the ECC to our dict for faster packing
+                block_data_hash = md5(block_data[: block_ecc_range.start]).digest()
+                DATA_HASHES[block_data_hash] = cur_block_ecc
+
+            # Check if there is data in the block
+            block_data_range = flash_attr.get_field_range_in_block(c, FlashFieldType.DATA)
+            if block_data_range is not None:
+                if block_ecc_range is not None:
+                    # Try decoding/correcting with ECC, otherwise just add the data anyway
+                    try:
+                        # Assumes that data comes before ECC
+                        if ecc_attr is not None and ecc_attr.ecc_class is not None:
+                            only_data += ecc_attr.ecc_class.decode(
+                                block_data[: block_ecc_range.end]
+                            )[block_data_range.start : block_data_range.end]
+                        else:
+                            raise UnpackerError(
+                                "Tried to correct with ECC without providing an ecc_class in F"
+                            )
+                    except EccError:
+                        only_data += block_data[block_data_range.start : block_data_range.end]
+                else:
+                    only_data += block_data[block_data_range.start : block_data_range.end]
+            offset += block_size
+
+        # Add all block data to logical resource for recursive unpacking
+        await ecc_resource.create_child(
+            tags=(FlashLogicalDataResource,),
+            data=only_data,
+            attributes=[
+                flash_attr,
+            ],
+        )
+        if ecc_attr is not None:
+            await ecc_resource.create_child(
+                tags=(FlashLogicalEccResource,),
+                data=only_ecc,
+                attributes=[
+                    ecc_attr,
+                ],
+            )
+
+
+#####################
+#      PACKERS      #
+#####################
+class FlashResourcePacker(Packer[None]):
+    """
+    Packs the FlashResource into binary and cleans up logical data representations
+    """
+
+    id = b"FlashResourcePacker"
+    targets = (FlashResource,)
+
+    async def pack(self, resource: Resource, config=None):
+        """
+        :param resource:
+        """
```

Comment:
If there is no docstring, the docs will use one for the superclass. But if there is a docstring, it will override the superclass one. 

This should either be filled out with a docstring or removed.

```suggestion
```